Mathlib/Analysis/SpecialFunctions/Gamma/Basic.lean

```diff
@@ -340,17 +341,24 @@ theorem Gamma_eq_integral {s : ℂ} (hs : 0 < s.re) : Gamma s = GammaIntegral s
   Gamma_eq_GammaAux s 0 (by norm_cast; linarith)
 #align complex.Gamma_eq_integral Complex.Gamma_eq_integral
 
-theorem Gamma_one : Gamma 1 = 1 := by rw [Gamma_eq_integral]; simpa using GammaIntegral_one; simp
+@[simp]
+theorem Gamma_one : Gamma 1 = 1 := by rw [Gamma_eq_integral]; simp; simp
 #align complex.Gamma_one Complex.Gamma_one
 
 theorem Gamma_nat_eq_factorial (n : ℕ) : Gamma (n + 1) = n ! := by
   induction' n with n hn
-  · simpa using Gamma_one
+  · simp
   · rw [Gamma_add_one n.succ <| Nat.cast_ne_zero.mpr <| Nat.succ_ne_zero n]
     simp only [Nat.cast_succ, Nat.factorial_succ, Nat.cast_mul]; congr
 #align complex.Gamma_nat_eq_factorial Complex.Gamma_nat_eq_factorial
 
+@[simp]
+theorem Gamma_ofNat_eq_factorial (n : ℕ) [(n + 1).AtLeastTwo] :
+    Gamma (no_index (OfNat.ofNat (n + 1) : ℂ)) = n ! := by
```

Comment:
I think the RHS here should have an OfNat

Reply:
The RHS is `(Nat.cast (n !) : ℂ)`. You're suggesting `OfNat.ofNat (n !)` instead? Then I get an error
```text
failed to synthesize instance
  OfNat ℂ (n !)
```
because there is no instance asserting `(n !).AtLeastTwo`

Reply:
I think the worry is that `simp` might simplify to a raw numeral. But if I simplify `Gamma 7` using this `simp`-lemma, I get `((OfNat.ofNat 6)! : ℂ)`, as desired.